source/local/local.go

```diff
@@ -144,17 +163,57 @@ func (s *localSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
-func parseFeatures(lines [][]byte) map[string]string {
+func parseDirectives(line string, fileName string, opts *parsingOpts) error {
```

Comment:
I think it would be just simpler to not care about the filename here. Just return an error and log it in the caller function.

```suggestion
func parseDirectives(line string, opts *parsingOpts) error {
```